PyDSS/pyPostprocessor/PostprocessScripts/EdLiFoControl.py

```diff
@@ -0,0 +1,595 @@
+        # "line_loading_limit",
+        # "dt_loading_limit",
+        # "line_safety_margin",
+        # "xfmr_safety_margin",
+        # "nominal_voltage",
+        # "max_iterations",
+        # "create_upgrade_plots",
+        # "tps_to_test",
+        # "create_upgrades_library",
+		# "upgrade_library_path",
+    # )
+
+    def __init__(self, project, scenario, inputs, dssInstance, dssSolver, dssObjects, dssObjectsByClass, simulationSettings, Logger):
+        """
+		Constructor method
+        """
+        super(EdLiFoControl, self).__init__(project, scenario, inputs, dssInstance, dssSolver, dssObjects, dssObjectsByClass, simulationSettings, Logger)
+        dss = dssInstance
+        #sol = dss.Solution
+        #self._export_path = self._dssPath['Export']
+        if isinstance(self.config["curtailment_size"], numbers.Number):
+            self.curtailment_size = self.config["curtailment_size"]
+        else:
+            self.curtailment_size = 20
+        
+        self.electrical_distance_file_path = self.config["electrical_distance_file_path"]
+        
+        if not os.path.exists(self.electrical_distance_file_path):
+            self.electrical_distance_file_path = os.path.join(self.config["Inputs"], "edistance.csv")
+            self.dismat_df = compute_electric_distance(dss)
+            self.dismat_df.to_csv(self.electrical_distance_file_path)
+            
+        else:
+            
+            self.dismat_df = pd.read_csv(self.electrical_distance_file_path, index_col=0)
+            
+        self.avg_distance, self.std_distance, self.dispersion_factor, self.zone_threshold = choose_zone_radius(self.dismat_df)
+        self.zone_option = self.config["zone_option"]
+        
+        if isinstance(self.config["zone_threshold"], numbers.Number):
+            self.zone_threshold = self.config["zone_threshold"]
+            
+        if os.path.exists(self.config["fixed_pv_path"]):
+            self.fixed_pv_path = self.config["fixed_pv_path"]
+        else:
+            self.fixed_pv_path = ""
+            
+        if os.path.exists(self.config["lifo_pv_path"]):
+            self.lifo_pv_path = self.config["lifo_pv_path"]
+        else:
+            self.lifo_pv_path = ""
+            
+        self.lifo_min_pv_size = self.config["lifo_min_pv_size"]
+        self.user_lifo_pv_list = self.config["user_lifo_pv_list"]
+        self.pvdict, self.pvdf = parse_pv_scenario(self.lifo_pv_path, self.lifo_min_pv_size)
+        self.pvdf3 = self.pvdf[self.pvdf['phases']=='3'].copy()
+        self.export_path = os.path.join(self.config["Outputs"])
+        self.monitored_lines = get_monitored_line_dataframe("all")
+        self.pctpmpp_step = 5
+        self.curtail_option = 'pctpmpp' # can be 'kva' or 'pctpmpp', default='pctpmpp'
+        
+        self.curtailment_record = {}
+        self.pvbus_voltage_record = {}
+        
+        
+        
+        
+        
+        
+    
+    def poa_curtail(self, dss):

```

Comment:
You imported opendssdirect at the top. No need to pass it here.

Reply:
Yeah, okay. Fixed